app/src/main/java/woowacourse/paint/MainActivity.kt

```diff
@@ -2,13 +2,69 @@ package woowacourse.paint
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.lifecycle.ViewModelProvider
+import com.google.android.material.slider.RangeSlider
+import woowacourse.paint.DrawingCanvas.Companion.DEFAULT_STROKE_WIDTH
+import woowacourse.paint.databinding.ActivityMainBinding
+import woowacourse.paint.model.ColorBox
 
+class MainActivity : AppCompatActivity(), OnColorClickListener {
 
-class MainActivity : AppCompatActivity() {
+    private val binding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val viewModel by lazy {
+        ViewModelProvider(this)[MainViewModel::class.java]
+    }
+
+    private val adapter by lazy {
+        ColorAdapter(this)
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContentView(binding.root)
+
+        setupRangeSlider()
+        setAdapter()
+        setListener()
+        setObserver()
+    }
+
+    private fun setupRangeSlider() {
+        binding.rangeSlider.setValues(DEFAULT_STROKE_WIDTH)
+    }
+
+    private fun setAdapter() {
+        binding.rvColor.adapter = adapter
+    }
+
+    private fun setListener() {
+        binding.rangeSlider.addOnChangeListener(
+            RangeSlider.OnChangeListener { _, value, _ ->
+                binding.cvPainter.setStroke(value)
+            },
+        )
+
+        binding.btnColorChange.setOnClickListener {
+            binding.rvColor.isVisible = !binding.rvColor.isVisible
+        }
+
+        binding.btnStrokeChange.setOnClickListener {
+            binding.rangeSlider.isVisible = !binding.rangeSlider.isVisible
+        }
+    }
+
+    private fun setObserver() {
+        viewModel.colors.observe(this) {
+            adapter.setColors(it)
```

Comment:
```suggestion
            adapter.submitList(it)
```
이 부분은 `submitList(it)`으로도 충분하다고 생각이 들고 있습니다!
추후 변경 시 code change를 줄이기 위해서 미리 만드신건가요??

Reply:
이 부분은 저도 아크 의견처럼 저렇게 submitList만 써도 될 것 같네요!! 
같은 역할을 하는 함수가 있는데 또 함수를 만들 필요는 없을 것 같아요😄 감사합니다👍